### PR TITLE
[BUG] Fix generic_assay_data/fetch empty selector filters returning server errors

### LINKED ISSUE #11588: [BUG] Generic Assay Data endpoints edge case when post body is empty
**Describe the bug**
When using post method: [/api/generic_assay_data/fetch](https://www.cbioportal.org/api/swagger-ui/index.html#/Generic%20Assay%20Data/fetchGenericAssayDataInMultipleMolecularProfilesUsingPOST), if both  sampleMolecularIdentifiers and molecularProfileIds are empty, the expection is not being hanlded correctly, catch and return bad request exception for this case

**Expected behavior**
When request body is not valid, returns 400 bad request

### PR DESCRIPTION
**Pull request by VarshiniGunti**
Fix #11588

Describe changes proposed in this pull request:
- Return `400 Bad Request` for `POST /api/generic_assay_data/fetch` when both `sampleMolecularIdentifiers` and `molecularProfileIds` are missing/empty.
- Add controller-side validation guard so invalid selector combinations do not reach service execution.
- Add regression tests in `GenericAssayDataControllerTest` for empty and missing selector cases.
- Harden `GlobalExceptionHandler#handleMethodArgumentNotValid` to safely handle object-level validation errors without field errors.

# Checks
- [x] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). We can fix this during merge by using a squash+merge if necessary
- [x] Has tests or has a separate issue that describes the types of test that should be created. If no test is included it should explicitly be mentioned in the PR why there is no test.
- [x] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!
- [ ] Make sure your PR has one of the labels defined in https://github.com/cBioPortal/cbioportal/blob/master/.github/release-drafter.yml

# Any screenshots or GIFs?
N/A (backend-only API validation and error handling fix; no UI changes)

### CODE REVIEW

**Comment by dippindots** [comment]
@VarshiniGunti the spotless check is failing, could you fix it with `mvn spotless:apply`

**Comment by VarshiniGunti** [comment]
> @VarshiniGunti the spotless check is failing, could you fix it with `mvn spotless:apply`

Hi, I have made the required chnages 
Can you please re-review it whenever you can
Thanks!

File: src/main/java/org/cbioportal/legacy/web/GenericAssayDataController.java [hunk 1]
```diff
@@ -228,4 +234,18 @@ private List<GenericAssayData> filterEmptyGenericAssayData(
         .filter(g -> StringUtils.isNotEmpty(g.getValue()) && !g.getValue().equals("NA"))
         .collect(Collectors.toList());
   }
+
+  private boolean hasValidMultipleStudyFilter(
```
Comment (dippindots) [inline src/main/java/org/cbioportal/legacy/web/GenericAssayDataController.java]:
```suggestion
  private boolean hasValidGenericAssayDataMultipleStudyFilter(
```

**Comment by dippindots** [comment]
@VarshiniGunti Had one comment about a variable name

**Comment by VarshiniGunti** [comment]
Hi @dippindots , I addressed your review comment by renaming
hasValidMultipleStudyFilter to hasValidGenericAssayDataMultipleStudyFilter in GenericAssayDataController.

I’ve pushed the update to this branch. Could you please re-review when you get a chance? Thanks!

**Comment by dippindots** [comment]
@VarshiniGunti it looks good now, could you remove this file `cbioportal-frontend` before we merge?
